libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt

```diff
@@ -195,18 +202,58 @@ class MapboxTripSessionTest {
         tripSession.stop()
     }
 
+    @Test
+    fun startSessionWithoutTripService() {
+        every { tripService.hasServiceStarted() } returns false
+
+        tripSession.start(false)
+
+        assertFalse(tripSession.isRunningWithForegroundService())
+        verify(exactly = 0) { tripService.startService() }
+        verify {
+            locationEngine.requestLocationUpdates(
+                any(),
+                any(),
+                Looper.getMainLooper()
+            )
+        }
+
+        tripSession.stop()
+    }
+
+    @Test
+    fun startSessionWithTripServiceCallStartAgain() {
+        tripSession.start(true)
+
+        tripSession.start(false)
+
+        every { tripService.hasServiceStarted() } returns true
```

Comment:
nit: this mock can be removed.

Reply:
If removed `assertTrue` will fail. Will let it be.

Reply:
:+1: I missed the assertion.